packages/remix-server-runtime/cookies.ts

```diff
@@ -32,7 +32,7 @@ export type CookieOptions = CookieParseOptions &
  *
  * @see https://remix.run/api/remix#cookie-api
  */
-export interface Cookie {
+export interface Cookie<Value extends unknown = unknown> {
```

Comment:
Let's keep `any` (for now), as it's causing some other type errors, which would bring us really far on this.
https://github.com/remix-run/remix/runs/6507483095?check_suite_focus=true#step:6:125

```suggestion
export interface Cookie<Value extends any = any> {
```